### ISSUE TITLE
Algolia support

### ISSUE THREAD

**Issue by kakawait** [kakawait/hugo-tranquilpeak-theme#8]

**Comment by kakawait** [comment]
Invest new feature from Hugo `0.20` http://gohugo.io/extras/output-formats/

**Comment by willwade** [comment]
We have recently got this working in our fork with a npm script:

see https://github.com/openassistive/hugo-tranquilpeak-theme/commit/525a0dc0ceccdc3796ebd57c3ca5e8e89ddaaec0

and https://github.com/openassistive/OpenATFrontEnd/commit/801776c5dd617da7931e716f5116adf5971beaf9

and to see it in practice: https://openassistive.org (click on search in nav). 

We run it with `npm run reindex || true && hugo` to build the site

**Comment by jhabdas** [comment]
https://www.algolia.com/pricing is a freemium tiered product. For those looking for search without paying for a product search can be added and scaled for free using open source tools such as Bloodhound by Twitter (or its logical successor) and https://www.lunrjs.com/.

If this were an e-commerce theme (which it's not), I would say the same thing.

**Comment by jhabdas** [comment]
https://gist.github.com/lgs/94eabc5993711268952ccde1234a9b20

> hugo + gruntjs + lunrjs = <3 search

**Comment by willwade** [comment]
Two problems I see with this:
1. The upstream project is using algolia and I know @kakawait wants to stick as closely as possible to that. Using a different JS library changes the project quite a bit
2. If you have a lot of data - using Lunr isn't a great idea.

**Comment by jhabdas** [comment]
Thanks for the info @willwade. Didn't realize the upstream was using Algolia. That's a shame IMO, as it leads individuals to rely on third parties when OSS can get them there free (one way or another), stymieing innovation as a result.

Regarding your second item, anyone having enough data to require something beyond Lunr probably isn't using this theme.

**Comment by willwade** [comment]
I agree on that @jhabdas  👍 - they probably aren't.. although we are (it may not be for long to be honest..). 

(Thanks for that gist by the way - I do love Lunr. I don't see a major problem by the way supporting it as well as algolia. do a fork ;) )

**Comment by kakawait** [comment]
About upstream project, I'm not open to add more feature on that theme that may not exist on upstream project, but I'm trying to report at least everything feature (that seems relevant) from upstream project. That why _Algolia_ support is tracked.

I'm sure `Lunr` I will check it but after _Algolia_, I can't work on both in same time (not enough time).

@willwade Thank for link I will check that. But as I said there is also new feature from _Hugo 0.20_ that should help.

**Comment by jhabdas** [comment]
> But as I said there is also new feature from Hugo 0.20 that should help.

And here it is:
https://gohugo.io/content/types/
